### PR TITLE
Add support to use "parent" hierarchy relationships with computed attributes

### PR DESCRIPTION
**Pull request by ogenstad**
This is to support computed attributes on hierarchial nodes. The "parent" relationship seems to work different compared to other relationships. This quick fix feels a bit hacky to me and we might want another long term solution with regards to how these relationships are defined and perhaps also look at refactoring the parse_schema_path() code and the SchemaAttributePath object.

I can add some additional tests for this but first wanted to check if anyone has any thoughts about alternate solutions for this just now.

### CODE REVIEW

File: backend/infrahub/support/macro.py [hunk 1]
```diff
@@ -58,7 +58,8 @@ def _parse_template(self) -> nodes.Template:
 
     def render(self, variables: dict[str, Any]) -> str:
         template = self.env.from_string(self.macro)
-        return template.render(variables)
+        output_dict = {key: ("" if value is None else value) for key, value in variables.items()}
+        return template.render(output_dict)
```
Comment (ogenstad) [inline backend/infrahub/support/macro.py]: This is to ensure that we use a blank string instead of `None` if a given value would be missing.

Comment (BeArchiTek) [reply backend/infrahub/support/macro.py]: We could still use None and do a default(omit) in the Jin’s depending if the use case

**Review by gmazoyer — APPROVED** [review]

File: backend/infrahub/support/macro.py (hunk 1 above)
Comment (ogenstad) [reply backend/infrahub/support/macro.py]: I reverted this part for now.

**Comment by ogenstad** [comment]
Rebased and added tests.

**Pull request merged**